apps/dashboard/app/javascript/icon_picker.js

```diff
@@ -64,20 +64,52 @@ function addSearch(){
   });
 }
 
-function searchIcons(event){
-  const searchString = event.target.value;
-  const rex = new RegExp(searchString, "g");
-  hiddenIcons = true;
+/**
+ * Creates an inverted index of ALL_ICONS.
+ * Returns an array type.
+ **/
+ function createInvertedIndex(arr) {
+  const invertedIndex = {};
+
+  arr.forEach((str, index) => {
+    for (let i = 0; i < str.length; i++) {
+      const char = str[i];
+      if (!invertedIndex[char]) {
+        invertedIndex[char] = [];
+      }
+      invertedIndex[char].push(index);
+    }
+  });
 
-  ALL_ICONS.forEach(name => {
+  return invertedIndex;
+}
+ 
+let invertedIndex = createInvertedIndex(ALL_ICONS);
+
+function searchIcons(event) {
+  const searchString = event.target.value.toLowerCase(); // Convert input to lowercase for case-insensitive search
+  const iconIds = Object.keys(invertedIndex); // Get all keys (characters) from the inverted index
+  const resultIndices = new Set(); // Set to store indices of matching icons
+
+  iconIds.forEach(char => {
+    const indices = invertedIndex[char]; // Get indices for the character in the inverted index
+    indices.forEach(index => {
+      const str = ALL_ICONS[index].toLowerCase();
+      if (str.includes(searchString)) {
```

Comment:
This is a little bit of a nit-pick, but can you change the variable `str` to `icon` or maybe `iconStr`?

That way `icon.includes(searchString)` reads a little better as to what the expression is trying to accomplish.